crates/proof-of-sql-parser/src/posql_time/unit.rs

```diff
@@ -53,10 +67,75 @@ impl fmt::Display for PoSQLTimeUnit {
 #[cfg(test)]
 #[allow(deprecated)]
 mod time_unit_tests {
-
-    use crate::posql_time::{timestamp::PoSQLTimestamp, unit::PoSQLTimeUnit};
+    use super::*;
+    use crate::{error::PoSQLTimestampError, posql_time::timestamp::PoSQLTimestamp};
     use chrono::{TimeZone, Utc};
 
+    #[test]
+    fn test_valid_precisions() {
+        assert_eq!(PoSQLTimeUnit::try_from("0"), Ok(PoSQLTimeUnit::Second));
+        assert_eq!(PoSQLTimeUnit::try_from("3"), Ok(PoSQLTimeUnit::Millisecond));
+        assert_eq!(PoSQLTimeUnit::try_from("6"), Ok(PoSQLTimeUnit::Microsecond));
+        assert_eq!(PoSQLTimeUnit::try_from("9"), Ok(PoSQLTimeUnit::Nanosecond));
+    }
+
+    #[test]
+    fn test_invalid_precisions() {
+        // Test some random incorrect values
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("1"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "1")
+        );
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("2"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "2")
+        );
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("4"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "4")
+        );
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("5"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "5")
+        );
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("7"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "7")
+        );
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("8"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "8")
+        );
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("10"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "10")
+        );
+        // Non-numeric strings
+        assert!(
+            matches!(PoSQLTimeUnit::try_from("abc"), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x == "abc")
+        );
+        // Empty string
+        assert!(
+            matches!(PoSQLTimeUnit::try_from(""), Err(PoSQLTimestampError::UnsupportedPrecision(x)) if x.is_empty())
```

Comment:
Please try one negative number as well

Reply:
I did but your feedback above seems to conflict with this